File: lib/ui/customers/customer_detail_page.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import '../../data/database/app_database.dart';
import '../../models/enums.dart';
import '../../providers/customer_providers.dart';

/// 客户详情页
class CustomerDetailPage extends ConsumerWidget {
  final String customerId;

  const CustomerDetailPage({super.key, required this.customerId});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final customerAsync = ref.watch(customerDetailProvider(customerId));
    final followUpsAsync = ref.watch(customerFollowUpsProvider(customerId));

    return PopScope(
      canPop: false,
      onPopInvokedWithResult: (didPop, _) {
        if (!didPop) context.go('/customers');
      },
      child: Scaffold(
      appBar: AppBar(
        title: Text(customerAsync.valueOrNull?.name ?? '客户详情'),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back),
          onPressed: () => context.go('/customers'),
        ),
        actions: [
          IconButton(
            icon: const Icon(Icons.edit_outlined),
            tooltip: '编辑',
            onPressed: () => context.push('/customer/$customerId/edit'),
          ),
        ],
      ),
      body: customerAsync.when(
        data: (customer) {
          if (customer == null) {
            return const Center(child: Text('客户不存在'));
          }
          return _DetailBody(
            customer: customer,
            followUpsAsync: followUpsAsync,
            onDelete: () => _confirmDelete(context, ref),
          );
        },
        loading: () => const Center(child: CircularProgressIndicator()),
        error: (e, _) => Center(child: Text('加载失败: $e')),
      ),
      ),
    );
  }

  Future<void> _confirmDelete(BuildContext context, WidgetRef ref) async {
    final confirmed = await showDialog<bool>(
      context: context,
      builder: (ctx) => AlertDialog(
        title: const Text('删除客户'),
        content: const Text('确认删除此客户？此操作不可撤销。'),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(ctx, false),
            child: const Text('取消'),
          ),
          TextButton(
            onPressed: () => Navigator.pop(ctx, true),
            child: const Text('删除', style: TextStyle(color: Colors.red)),
          ),
        ],
      ),
    );
    if (confirmed == true) {
      await ref.read(deleteCustomerProvider(customerId).future);
      if (!context.mounted) return;
      context.go('/customers');
    }
  }
}

/// 详情内容
class _DetailBody extends StatelessWidget {
  final CustomerEntity customer;
  final AsyncValue<List<FollowUpEntity>> followUpsAsync;
  final VoidCallback onDelete;

  const _DetailBody({
    required this.customer,
    required this.followUpsAsync,
    required this.onDelete,
  });

  @override
  Widget build(BuildContext context) {
    final operator = Operator.fromCode(customer.operator);
    final stage = CustomerStage.fromCode(customer.salesStage);
    final hasPackage = customer.actualCost != null ||
        (customer.packageName != null && customer.packageName!.isNotEmpty) ||
        (customer.traffic != null && customer.traffic!.isNotEmpty) ||
        (customer.minutes != null && customer.minutes!.isNotEmpty) ||
        customer.broadband ||
        customer.camera ||
        customer.subCards > 0;

    return ListView(
      padding: const EdgeInsets.fromLTRB(16, 8, 16, 24),
      children: [
        // === 基础信息卡片 ===
        _Card(
          title: '基础信息',
          children: [
            _InfoRow(label: '称呼', value: customer.name),
            if (customer.phone.isNotEmpty)
              _InfoRow(label: '手机号', value: customer.phone),
            _InfoRow(label: '运营商', value: operator.label),
            _InfoRow(
              label: '月消费',
              value: _costLabel(customer.selfReportedCost),
            ),
            _InfoRow(label: '状态', value: stage.label),
          ],
        ),
        const SizedBox(height: 12),

        // === 套餐详情卡片 (如果有) ===
        if (hasPackage) ...[
          _Card(
            title: '套餐详情',
            children: [
              if (customer.packageName != null &&
                  customer.packageName!.isNotEmpty)
                _InfoRow(label: '套餐名称', value: customer.packageName!),
              if (customer.actualCost != null)
                _InfoRow(label: '实际消费', value: '${customer.actualCost}元'),
              if (customer.traffic != null && customer.traffic!.isNotEmpty)
                _InfoRow(label: '流量', value: customer.traffic!),
              if (customer.minutes != null && customer.minutes!.isNotEmpty)
                _InfoRow(label: '通话', value: customer.minutes!),
              _InfoRow(label: '宽带', value: customer.broadband ? '有' : '无'),
              _InfoRow(label: '摄像头', value: customer.camera ? '有' : '无'),
              _InfoRow(
                label: '副卡',
                value: customer.subCards > 0
                    ? '${customer.subCards}张'
                    : '无',
              ),
            ],
          ),
          const SizedBox(height: 12),
        ],

        // === 备注区域 ===
        if (customer.note != null && customer.note!.isNotEmpty) ...[
          _Card(
            title: '备注',
            children: [
              Text(
                customer.note!,
                style: Theme.of(context).textTheme.bodyMedium,
              ),
            ],
          ),
          const SizedBox(height: 12),
        ],

        // === 跟进记录区域 ===
        _Card(
          title: '跟进记录',
          children: [
            followUpsAsync.when(
              data: (followUps) {
                if (followUps.isEmpty) {
                  return const Padding(
                    padding: EdgeInsets.symmetric(vertical: 8),
                    child: Text('暂无跟进记录',
                        style: TextStyle(color: Colors.grey)),
                  );
                }
                return Column(
                  children: followUps
                      .map((f) => _FollowUpTile(followUp: f))
                      .toList(),
                );
              },
              loading: () => const Padding(
                padding: EdgeInsets.symmetric(vertical: 8),
                child: Center(
                  child: SizedBox(
                    width: 20,
                    height: 20,
                    child: CircularProgressIndicator(strokeWidth: 2),
                  ),
                ),
              ),
              error: (_, __) =>
                  const Text('加载失败', style: TextStyle(color: Colors.grey)),
            ),
          ],
        ),
        const SizedBox(height: 24),

        // === 底部: 删除客户按钮 ===
        OutlinedButton.icon(
          onPressed: onDelete,
          icon: const Icon(Icons.delete_outline, color: Colors.red),
          label: const Text('删除客户', style: TextStyle(color: Colors.red)),
          style: OutlinedButton.styleFrom(
            side: const BorderSide(color: Colors.red),
            minimumSize: const Size(double.infinity, 52),
          ),
        ),
      ],
    );
  }

  String _costLabel(int? cost) => cost == null ? '不清楚' : '$cost元';
}

/// 跟进记录条目
class _FollowUpTile extends StatelessWidget {
  final FollowUpEntity followUp;
  const _FollowUpTile({required this.followUp});

  @override
  Widget build(BuildContext context) {
    final dt = followUp.scheduledAt;
    final timeStr =
        '${dt.month}/${dt.day} ${dt.hour.toString().padLeft(2, '0')}:${dt.minute.toString().padLeft(2, '0')}';

    return ListTile(
      dense: true,
      contentPadding: EdgeInsets.zero,
      leading: Icon(
        followUp.completed ? Icons.check_circle : Icons.schedule,
        color: followUp.completed
            ? Colors.green
            : Theme.of(context).colorScheme.primary,
        size: 22,
      ),
      title: Text(
        followUp.content ?? '跟进',
        style: TextStyle(
          decoration:
              followUp.completed ? TextDecoration.lineThrough : null,
          color: followUp.completed ? Colors.grey : null,
        ),
      ),
      subtitle: Text(timeStr),
    );
  }
}

/// 信息卡片
class _Card extends StatelessWidget {
  final String title;
  final List<Widget> children;
  const _Card({required this.title, required this.children});

  @override
  Widget build(BuildContext context) {
    return Card(
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              title,
              style: Theme.of(context)
                  .textTheme
                  .titleSmall
                  ?.copyWith(fontWeight: FontWeight.bold),
            ),
            const SizedBox(height: 8),
            ...children,
          ],
        ),
      ),
    );
  }
}

/// 信息行 (label : value)
class _InfoRow extends StatelessWidget {
  final String label;
  final String value;
  const _InfoRow({required this.label, required this.value});

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 4),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(
            label,
            style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                  color: Theme.of(context).colorScheme.onSurfaceVariant,
                ),
          ),
          Flexible(
            child: Text(
              value,
              style: const TextStyle(fontWeight: FontWeight.w600),
              textAlign: TextAlign.right,
            ),
          ),
        ],
      ),
    );
  }
}

File: app/src/main/java/com/salesquest/sales_quest/ui/customers/CustomerDetailPage.kt
package com.salesquest.sales_quest.ui.customers

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.DeleteOutline
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Scaffold
import androidx.compose.material3.SnackbarHost
import androidx.compose.material3.SnackbarHostState
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.salesquest.sales_quest.core.AppContainer
import com.salesquest.sales_quest.data.CustomerStage
import com.salesquest.sales_quest.data.DateUtil
import com.salesquest.sales_quest.data.Operator
import com.salesquest.sales_quest.data.entity.CustomerEntity
import kotlinx.coroutines.launch
import java.text.SimpleDateFormat
import java.util.Locale

/** 客户详情页 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CustomerDetailPage(
    customerId: String,
    onBack: () -> Unit = {},
    onEdit: (String) -> Unit = {},
    viewModel: CustomerDetailViewModel = viewModel(factory = CustomerDetailViewModel.factory(customerId))
) {
    val customer by viewModel.customer.collectAsState()
    val followUps by viewModel.followUps.collectAsState()
    val snackbarHostState = remember { SnackbarHostState() }
    val scope = rememberCoroutineScope()
    var showDeleteConfirm by remember { mutableStateOf(false) }

    Scaffold(
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text(customer?.name ?: "客户详情") },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "返回")
                    }
                },
                actions = {
                    IconButton(onClick = { onEdit(customerId) }) {
                        Icon(Icons.Filled.Edit, contentDescription = "编辑")
                    }
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors()
            )
        },
        snackbarHost = { SnackbarHost(snackbarHostState) }
    ) { innerPadding ->
        if (customer == null) {
            Box(Modifier.fillMaxSize().padding(innerPadding), contentAlignment = Alignment.Center) {
                Text("客户不存在")
            }
        } else {
            val c = customer!!
            DetailBody(
                customer = c,
                followUps = followUps,
                onDelete = { showDeleteConfirm = true },
                modifier = Modifier.padding(innerPadding)
            )
        }
    }

    if (showDeleteConfirm) {
        AlertDialog(
            onDismissRequest = { showDeleteConfirm = false },
            title = { Text("删除客户") },
            text = { Text("确认删除此客户？此操作不可撤销。") },
            confirmButton = {
                TextButton(
                    onClick = {
                        showDeleteConfirm = false
                        scope.launch {
                            try {
                                AppContainer.deleteCustomer(customerId)
                                onBack()
                            } catch (e: Exception) {
                                snackbarHostState.showSnackbar("删除失败: ${e.message}")
                            }
                        }
                    }
                ) { Text("删除", color = Color.Red) }
            },
            dismissButton = {
                TextButton(onClick = { showDeleteConfirm = false }) { Text("取消") }
            }
        )
    }
}

/** 详情内容 */
@Composable
fun DetailBody(
    customer: CustomerEntity,
    followUps: List<com.salesquest.sales_quest.data.entity.FollowUpEntity>,
    onDelete: () -> Unit,
    modifier: Modifier = Modifier
) {
    val operator = Operator.fromCode(customer.operator)
    val stage = CustomerStage.fromCode(customer.salesStage)
    val hasPackage = customer.actualCost != null ||
        (customer.packageName != null && customer.packageName.isNotEmpty()) ||
        (customer.traffic != null && customer.traffic.isNotEmpty()) ||
        (customer.minutes != null && customer.minutes.isNotEmpty()) ||
        customer.broadband || customer.camera || customer.subCards > 0

    LazyColumn(
        modifier = modifier.fillMaxSize(),
        contentPadding = androidx.compose.foundation.layout.PaddingValues(start = 16.dp, top = 8.dp, end = 16.dp, bottom = 24.dp)
    ) {
        item {
            InfoCard(title = "基础信息") {
                InfoRow("称呼", customer.name)
                if (customer.phone.isNotEmpty()) InfoRow("手机号", customer.phone)
                InfoRow("运营商", operator.label)
                InfoRow("月消费", costLabel(customer.selfReportedCost))
                InfoRow("状态", stage.label)
            }
        }

        if (hasPackage) {
            item {
                Spacer(Modifier.height(12.dp))
                InfoCard(title = "套餐详情") {
                    if (customer.packageName != null && customer.packageName.isNotEmpty()) {
                        InfoRow("套餐名称", customer.packageName)
                    }
                    if (customer.actualCost != null) InfoRow("实际消费", "${customer.actualCost}元")
                    if (customer.traffic != null && customer.traffic.isNotEmpty()) InfoRow("流量", customer.traffic)
                    if (customer.minutes != null && customer.minutes.isNotEmpty()) InfoRow("通话", customer.minutes)
                    InfoRow("宽带", if (customer.broadband) "有" else "无")
                    InfoRow("摄像头", if (customer.camera) "有" else "无")
                    InfoRow("副卡", if (customer.subCards > 0) "${customer.subCards}张" else "无")
                }
            }
        }

        if (customer.note != null && customer.note.isNotEmpty()) {
            item {
                Spacer(Modifier.height(12.dp))
                InfoCard(title = "备注") {
                    Text(customer.note, style = MaterialTheme.typography.bodyMedium)
                }
            }
        }

        item {
            Spacer(Modifier.height(12.dp))
            InfoCard(title = "跟进记录") {
                if (followUps.isEmpty()) {
                    Text("暂无跟进记录", color = Color.Gray, modifier = Modifier.padding(vertical = 8.dp))
                } else {
                    followUps.forEach { f ->
                        FollowUpTile(followUp = f)
                    }
                }
            }
        }

        item {
            Spacer(Modifier.height(24.dp))
            OutlinedButton(
                onClick = onDelete,
                modifier = Modifier.fillMaxWidth().height(52.dp)
            ) {
                Icon(Icons.Filled.DeleteOutline, contentDescription = null, tint = Color.Red)
                Spacer(Modifier.size(8.dp))
                Text("删除客户", color = Color.Red)
            }
        }
    }
}

@Composable
fun FollowUpTile(followUp: com.salesquest.sales_quest.data.entity.FollowUpEntity) {
    val timeStr = formatTime(followUp.scheduledAt)
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            if (followUp.completed) Icons.Filled.CheckCircle else Icons.Filled.Schedule,
            contentDescription = null,
            tint = if (followUp.completed) Color(0xFF4CAF50) else MaterialTheme.colorScheme.primary,
            modifier = Modifier.size(22.dp)
        )
        Spacer(Modifier.size(8.dp))
        Column(modifier = Modifier.weight(1f)) {
            Text(
                followUp.content ?: "跟进",
                style = MaterialTheme.typography.bodyMedium,
                textDecoration = if (followUp.completed) androidx.compose.ui.text.style.TextDecoration.LineThrough else null,
                color = if (followUp.completed) Color.Gray else Color.Unspecified
            )
            Text(timeStr, style = MaterialTheme.typography.labelSmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
        }
    }
}

@Composable
fun InfoCard(title: String, content: @Composable ColumnScope.() -> Unit) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(16.dp)) {
            Text(title, style = MaterialTheme.typography.titleSmall, fontWeight = FontWeight.Bold)
            Spacer(Modifier.height(8.dp))
            content()
        }
    }
}

@Composable
fun InfoRow(label: String, value: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp),
        horizontalArrangement = androidx.compose.foundation.layout.Arrangement.SpaceBetween
    ) {
        Text(label, style = MaterialTheme.typography.bodyMedium, color = MaterialTheme.colorScheme.onSurfaceVariant)
        Text(value, style = MaterialTheme.typography.bodyMedium, fontWeight = FontWeight.SemiBold)
    }
}

private fun costLabel(cost: Int?): String = if (cost == null) "不清楚" else "${cost}元"

private fun formatTime(ts: Long): String {
    val sdf = SimpleDateFormat("M/d HH:mm", Locale.getDefault())
    return sdf.format(java.util.Date(ts))
}
